Quarkus v3.31.2: Jacoco unable to find the project root

### Describe the bug

After upgrading a multi-module project to v3.31.2 (or .3 and .4), tests fail to execute because the JacocoProcessor throws an error "unable to find the project root".

https://github.com/quarkusio/quarkus/blob/90bae5bb14ca0f765f2dec84f7c2d46e45f4e471/test-framework/jacoco/deployment/src/main/java/io/quarkus/jacoco/deployment/JacocoProcessor.java#L75

### Expected behavior

It works as expected.

### Actual behavior

An exception is thrown.

### How to Reproduce?

I'm writing this on mobile, I will edit this with a stackTrace/reproduced later.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

Seems related to #44013.

I've created a *very rough* reproducer [here](https://github.com/rgmz/quarkus-52989).
```sh
$ git clone git@github.com:rgmz/quarkus-52989.git
$ cd quarkus-52989
$ mvn clean test -pl app -am

[ERROR] Caused by: java.lang.reflect.UndeclaredThrowableException
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:904)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:1583)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: java.nio.file.NoSuchFileException: /var/tmp/code-with-quarkus2/code-with-quarkus/target/report-sources.txt
[ERROR]         at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
[ERROR]         at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
[ERROR]         at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
[ERROR]         at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:261)
[ERROR]         at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:482)
[ERROR]         at java.base/java.nio.file.Files.newOutputStream(Files.java:228)
[ERROR]         at java.base/java.nio.file.Files.write(Files.java:3574)
[ERROR]         at java.base/java.nio.file.Files.write(Files.java:3626)
[ERROR]         at io.quarkus.jacoco.deployment.JacocoProcessor.transform(JacocoProcessor.java:244)
```
https://github.com/quarkusio/quarkus/blob/85c542881d0dfc5fa95609159e78f065edbbd617/test-framework/jacoco/deployment/src/main/java/io/quarkus/jacoco/deployment/JacocoProcessor.java#L243

------

This produces a slightly different error on v3.31.3 and v3.31.4 but the underlying cause is similar. These settings are from the [Jacoco guide for multi-module plugins](https://quarkus.io/guides/tests-with-coverage).
```
      <quarkus.jacoco.data-file>${maven.multiModuleProjectDirectory}/target/jacoco.exec</quarkus.jacoco.data-file>
      <quarkus.jacoco.reuse-data-file>true</quarkus.jacoco.reuse-data-file>
      <quarkus.jacoco.report-location>${maven.multiModuleProjectDirectory}/target/coverage</quarkus.jacoco.report-location>
```

This doesn't occur if an empty top-level `target/` directory exists.
```sh
$ mvn clean && mkdir target/ && mvn test -pl app -am
...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for code-with-quarkus 1.0.0-SNAPSHOT:
[INFO] 
[INFO] code-with-quarkus .................................. SUCCESS [  0.002 s]
[INFO] feature1 ........................................... SUCCESS [  3.084 s]
[INFO] feature2 ........................................... SUCCESS [  0.225 s]
[INFO] app ................................................ SUCCESS [ 18.858 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  23.439 s
[INFO] Finished at: 2026-03-13T13:11:56-04:00
[INFO] ------------------------------------------------------------------------
```

--- a/test-framework/jacoco/deployment/src/main/java/io/quarkus/jacoco/deployment/JacocoProcessor.java
+++ b/test-framework/jacoco/deployment/src/main/java/io/quarkus/jacoco/deployment/JacocoProcessor.java
@@ -83,6 +83,9 @@ public class JacocoProcessor {
         } else {
             dataFilePath = outputDir.resolve(JacocoConfig.JACOCO_QUARKUS_EXEC);
         }
+        // Make sure the parent directory of the data file exists
+        Files.createDirectories(dataFilePath.getParent());
+
         String dataFile = dataFilePath.toString();
         log.debugf("JaCoCo destFile: %s", dataFilePath);
 

